spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/SparkReadConf.java

```diff
@@ -67,14 +62,13 @@ public boolean caseSensitive() {
   }
 
   public boolean localityEnabled() {
-    if (table.io() instanceof HadoopFileIO) {
-      HadoopInputFile file = (HadoopInputFile) table.io().newInputFile(table.location());
-      String scheme = file.getFileSystem().getScheme();
-      boolean defaultValue = LOCALITY_WHITELIST_FS.contains(scheme);
+    if (Util.usesHadoopFileIO(table.io(), table.location())) {
```

Comment:
Can this be simplified like this?

```
boolean defaultValue = Util.mayHaveBlockLocations(table.io(), table.location());
return PropertyUtil.propertyAsBoolean(readOptions, SparkReadOptions.LOCALITY, defaultValue);
```

Reply:
+1, this would match the flink behaviour, i refrained for making them same initially because let's say if my FileIO is S3FileIO and from spark read option i pass locality as true, I would make locality enabled as false and unnecessarily check `mayHaveBlockLocations` earlier we would have short circuited it and made locality enabled as false.

Reply:
@singhpk234, I think `Util.mayHaveBlockLocations` internally calls `usesHadoopFileIO` so I am not sure I got your point [here](https://github.com/apache/iceberg/pull/6655#discussion_r1095306379).